cmd/mistryd/server.go

```diff
@@ -379,34 +391,73 @@ func RebuildImages(cfg *Config, log *log.Logger, out io.Writer, projects []strin
 
 	ctx := context.Background()
 	for _, project := range projects {
+		start := time.Now()
 		log.Printf("Rebuilding %s...\n", project)
 		j, err := NewJob(project, types.Params{}, "", cfg)
 		if err != nil {
-			r.failed++
+			r.failed = append(r.failed, project)
 			log.Printf("Failed to instantiate %s job with error: %s\n", project, err)
 			if failOnImageError {
 				return r, err
 			}
 		} else {
-			err = j.BuildImage(ctx, cfg.UID, client, out, true, true)
-			if err != nil {
-				r.failed++
-				log.Printf("Failed to build %s job %s with error: %s\n", project, j.ID, err)
+			var buildErr error
+			if verbose {
+				// pipe image build logs to the logger
+				pr, pw := io.Pipe()
+				buildResult := make(chan error)
+
+				go func() {
+					err := j.BuildImage(ctx, cfg.UID, client, pw, true, true)
+					pw.Close()
```

Comment:
What about the error?

Reply:
its returned as-is through the channel to the main thread, which takes care of dealing with it (fail early or append to list of failures)

Reply:
I mean, the error from `pw.Close()`

Reply:
aha!

Reply:
i looked into the implementation of PipeWriter.Close(), it *never* returns an error even if the function defines it as such. I will add a check here and just log the error if not nil and proceed. The error will always be nil at least with Go 1.10

Reply:
Cool.